storage/imxrt-flash/flashsrv.c

```diff
@@ -559,32 +569,46 @@ static void flashsrv_devThread(void *arg)
 		}
 
 		mutexLock(memory->lock);
-
-		switch (msg.type) {
-			case mtRead:
-				msg.o.err = flashsrv_bufferedRead(memory->fOid.id, msg.i.io.offs, msg.o.data, msg.o.size);
-				break;
-
-			case mtWrite:
-				msg.o.err = flashsrv_bufferedWrite(memory->fOid.id, msg.i.io.offs, msg.i.data, msg.i.size);
-				break;
-
-			case mtDevCtl:
-				flashsrv_devCtl(memory, &msg);
-				break;
-
-			case mtOpen:
-			case mtClose:
-				msg.o.err = EOK;
-				break;
-
-			case mtSync:
-				msg.o.err = flashsrv_bufferedSync(memory->fOid.id);
-				break;
-
-			default:
-				msg.o.err = -ENOSYS;
-				break;
+		if (msg.oid.id >= memory->pCnt) {
```

Comment:
I think you mixed up the threads (here we base on `memory->fOid.id`) and in `flashsrv_rawThread` the id is read from msg.

Reply:
Right, thank you